packages/cozy-client/src/CozyClient.spec.js

```diff
@@ -1402,6 +1401,43 @@ describe('CozyClient', () => {
         receiveQueryResult('allTodos', fakeResponse)
       )
     })
+
+    const setupOnError = () => {
+      const error = new TypeError('Failed to fetch')
+      requestHandler.mockReturnValueOnce(Promise.reject(error))
+      const onError = jest.fn().mockReturnValueOnce(() => true)
+      return onError
+    }
+
+    it('should call onError callback when an error is thrown. Callback passed as argument', async () => {
+      const onError = setupOnError()
+
+      await client.query(query, { as: 'allTodos', onError })
+      expect(onError).toBeCalled()
+    })
+
+    it('should call onError callback when an error is catched. It is passed during instantiation', async () => {
+      const onError = setupOnError()
+      const client = new CozyClient({ onError })
+
+      await client.query(query, { as: 'allTodos' })
+      expect(onError).toBeCalled()
+    })
+
+    it('should call onError callback when an error is catched. It is passed with setOnError', async () => {
+      const onError = setupOnError()
+      client.setOnError(onError)
+
+      await client.query(query, { as: 'allTodos' })
+      expect(onError).toBeCalled()
+    })
+
+    it('should thrown an error when there is no onError callback', async () => {
+      try {
+        await client.query(query, { as: 'allTodos' })
+      } catch (e) {} // eslint-disable-line no-empty
+      await client.query(query, { as: 'allTodos' })
```

Comment:
I don't understand this test? 

await expect(client.query(query, { as: 'allTodos' })).rejects.toThrow();

Reply:
Oops, an oversight